src/_P163_RadSens.ino

```diff
@@ -0,0 +1,150 @@
+#include "_Plugin_Helper.h"
+#ifdef USES_P163
+
+// #######################################################################################################
+// ########################## Plugin 163: Counter - RadSens I2C radiation counter ########################
+// #######################################################################################################
+
+/** Changelog:
+ * 2024-08-12 tonhuisman: Start plugin for RadSens I2C radiation counter using RadSens library
+ * (Newest changes on top)
+ **/
+
+/** Commands:
+ * radsens,calibration,<calibrationvalue> : Set new Calibration value in impulses per millirad. Default 105 imp/uR.
+ */
+
+# define PLUGIN_163
+# define PLUGIN_ID_163          163
+# define PLUGIN_NAME_163        "Counter - RadSens I2C radiation counter"
+# define PLUGIN_VALUENAME1_163  "Count"
+# define PLUGIN_VALUENAME2_163  "iDynamic"
+# define PLUGIN_VALUENAME3_163  "iStatic"
+
+# include "./src/PluginStructs/P163_data_struct.h"
+
+boolean Plugin_163(uint8_t function, struct EventStruct *event, String& string)
+{
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+      Device[++deviceCount].Number         = PLUGIN_ID_163;
+      Device[deviceCount].Type             = DEVICE_TYPE_I2C;
+      Device[deviceCount].VType            = Sensor_VType::SENSOR_TYPE_SINGLE;
+      Device[deviceCount].Ports            = 0;
+      Device[deviceCount].FormulaOption    = true;
+      Device[deviceCount].ValueCount       = 3;
+      Device[deviceCount].SendDataOption   = true;
+      Device[deviceCount].TimerOption      = true;
+      Device[deviceCount].TimerOptional    = true;
+      Device[deviceCount].GlobalSyncOption = true;
+      Device[deviceCount].PluginStats      = true;
+
+      break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      string = F(PLUGIN_NAME_163);
+
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_163));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[1], PSTR(PLUGIN_VALUENAME2_163));
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[2], PSTR(PLUGIN_VALUENAME3_163));
+
+      break;
+    }
+
+    case PLUGIN_I2C_HAS_ADDRESS:
+    {
+      success = (RS_DEFAULT_I2C_ADDRESS == event->Par1);
+
+      break;
+    }
+
+    # if FEATURE_I2C_GET_ADDRESS
+    case PLUGIN_I2C_GET_ADDRESS:
+    {
+      event->Par1 = RS_DEFAULT_I2C_ADDRESS;
+      success     = true;
+      break;
+    }
+    # endif // if FEATURE_I2C_GET_ADDRESS
+
+    case PLUGIN_SET_DEFAULTS:
+    {
+      P163_SET_LED_STATE(true);                                    // Device defaults
+      P163_SET_LOW_POWER(false);
+      P163_CFG_THRESHOLD                         = -1;             // Threshold disabled
+      Settings.TaskDeviceTimer[event->TaskIndex] = Settings.Delay; // Set default like non-TimerOptional
```

Comment:
Why is this needed here? We don't do that in any other plugin.
If this is needed, maybe we should do this before calling this `PLUGIN_SET_DEFAULTS` ?

Reply:
I didn't want the default Interval to be 0 for this plugin, but still allow it to be set to 0. And this is the default value for plugins that don't allow an interval of 0.